src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -420,8 +500,28 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
                     examStatistics.medianGradeTotal = this.gradingSystemService.findMatchingGradeStep(this.gradingScale!.gradeSteps, examStatistics.medianRelativeTotal)!.gradeName;
                 }
             }
+            // Calculate total statistics if second correction exists
+            if (this.hasSecondCorrectionAndStarted) {
+                examStatistics.meanPointsTotalInFirstCorrection = SimpleStatistics.mean(studentPointsTotalInFirstCorrectionRound);
+                examStatistics.medianTotalInFirstCorrection = SimpleStatistics.median(studentPointsTotalInFirstCorrectionRound);
+                examStatistics.standardDeviationTotalInFirstCorrection = SimpleStatistics.standardDeviation(studentPointsTotalInFirstCorrectionRound);
+                if (this.examScoreDTO.maxPoints) {
+                    examStatistics.meanPointsRelativeTotalInFirstCorrection = (examStatistics.meanPointsTotalInFirstCorrection / this.examScoreDTO.maxPoints) * 100;
+                    examStatistics.medianRelativeTotalInFirstCorrection = (examStatistics.medianTotalInFirstCorrection / this.examScoreDTO.maxPoints) * 100;
+                    if (this.gradingScaleExists) {
+                        examStatistics.meanGradeTotalInFirstCorrection = this.gradingSystemService.findMatchingGradeStep(
+                            this.gradingScale!.gradeSteps,
+                            examStatistics.meanPointsRelativeTotalInFirstCorrection,
+                        )!.gradeName;
+                        examStatistics.medianGradeTotalInFirstCorrection = this.gradingSystemService.findMatchingGradeStep(
+                            this.gradingScale!.gradeSteps,
+                            examStatistics.medianRelativeTotalInFirstCorrection,
+                        )!.gradeName;
+                    }
+                }
+            }
```

Comment:
Isn't this part exactly repeated three times? It would be good if we can have a single helper function if that is the case.

Reply:
We have got 3 different cases here, one considers only passed exams, one only submitted exams and one considers all exams. In each of those cases the approach is the same, but we do access and define different attributes in each of those cases. I don't think we can summarise those, can we?

Reply:
Ok, inserting variables into the string doesn't look better, so we can leave it as is.